exasol/toolbox/util/release/changelog.py

```diff
@@ -2,42 +2,48 @@
 
 from datetime import datetime
 from inspect import cleandoc
+from itertools import chain
 from pathlib import Path
 
+from exasol.toolbox.util.dependencies.poetry_dependencies import (
+    get_dependencies,
+    get_dependencies_from_latest_tag,
+)
+from exasol.toolbox.util.dependencies.track_changes import DependencyChanges
 from exasol.toolbox.util.version import Version
 
 UNRELEASED_INITIAL_CONTENT = "# Unreleased\n"
 
 
 class Changelogs:
-    def __init__(self, changes_path: Path, version: Version) -> None:
+    def __init__(self, changes_path: Path, source_path: Path, version: Version) -> None:
         self.version = version
         self.unreleased_md: Path = changes_path / "unreleased.md"
         self.versioned_changelog_md: Path = changes_path / f"changes_{version}.md"
         self.changelog_md: Path = changes_path / "changelog.md"
+        self.source_path: Path = source_path
 
     def _create_new_unreleased(self):
         """
         Write a new unreleased changelog file.
         """
         self.unreleased_md.write_text(UNRELEASED_INITIAL_CONTENT)
 
-    def _create_versioned_changelog(self, content: str) -> None:
+    def _create_versioned_changelog(self, unreleased_content: str) -> None:
         """
         Create a changelog entry for a specific version.
 
         Args:
-            content: The content of the changelog entry.
+            unreleased_content: The content of the changelog entry.
```

Comment:
We have a terminology problem here.
Proposal: we could call 

- directory `doc/changes` to contain the whole (abstract) _change log_ or _log of changes_
- `changelog.md` to be the _index_ or _inventory_ of the _change log_ 
- `unreleased.md`:  to be a _changes file_
- `changes_1.6.1.md` to be a (versioned) _changes file_